src/v/archival/segment_reupload.cc

```diff
@@ -327,13 +327,24 @@ segment_collector::make_upload_candidate(
       first->index().base_timestamp(),
       io_priority_class);
 
+    if (head_seek_result.error()) {
+        co_return upload_candidate_with_locks{upload_candidate{}, {}};
```

Comment:
Would be nice to use a std::optional and return nullopt rather than default-unitialized upload_candidate